test/cli/harness/harness.go

```diff
@@ -188,3 +190,16 @@ func (h *Harness) Cleanup() {
 		log.Panicf("removing temp dir %s: %s", h.Dir, err)
 	}
 }
+
+// ExtractPeerID extracts a peer ID from the given multiaddr, and fatals if it does not contain a peer ID.
+func (h *Harness) ExtractPeerID(m multiaddr.Multiaddr) peer.ID {
+	peerIDStr, err := m.ValueForProtocol(multiaddr.P_P2P)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
This will incorrectly select the relay p2p's peer id in relayed addresses.
That because it does left to right matching instead of right to left.
I guess this is an edge case, but I would do:
```suggestion
	var peerIDStr string
	multiaddr.ForEach(m, func(c multiaddr.Component) bool {
	  if c.Protocol() != multiaddr.P_P2P {
	    return true
	  }
	  
	  peerIDStr = c.Value()
	  
	  return true
	})
	
	if peerIDStr == "" {
	  panic(multiaddr.ErrProtocolNotFound)
	}
```